blockindex/liquidstaking_bucket.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2023 IoTeX Foundation
+// This source code is provided 'as is' and no warranties are given as to title or non-infringement, merchantability
+// or fitness for purpose and, to the extent permitted by law, all liability for your use of the code is disclaimed.
+// This source code is governed by Apache License 2.0 that can be found in the LICENSE file.
+
+package blockindex
+
+import (
+	"math"
+	"math/big"
+
+	"github.com/iotexproject/iotex-address/address"
+	"github.com/pkg/errors"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/iotexproject/iotex-core/blockindex/indexpb"
+	"github.com/iotexproject/iotex-core/pkg/util/byteutil"
+)
+
+const (
+	maxBlockNumber = math.MaxUint64
+)
+
+type (
+	// ContractStakingBucketInfo is the bucket information
+	ContractStakingBucketInfo struct {
+		TypeIndex  uint64
+		CreatedAt  uint64
+		UnlockedAt uint64
+		UnstakedAt uint64
+		Delegate   address.Address // owner address of the delegate
+		Owner      address.Address
+	}
+
+	// ContractStakingBucketType is the bucket type
+	ContractStakingBucketType struct {
+		Amount      *big.Int
+		Duration    uint64
+		ActivatedAt uint64
+	}
+
+	// ContractStakingBucket is the bucket information including bucket type and bucket info
+	ContractStakingBucket struct {
+		Index                     uint64
+		Candidate                 address.Address
+		Owner                     address.Address
```

Comment:
question: different values with those in bucketinfo?

Reply:
the `ContractStakingBucket` is corresponding to `staking.VoteBucket`, `type ContractStakingBucket = staking.VoteBucket` will be added in pr 3860